lib/vrl/stdlib/src/to_regex.rs

```diff
@@ -0,0 +1,75 @@
+use regex::Regex;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToRegex;
+
+impl Function for ToRegex {
+    fn identifier(&self) -> &'static str {
+        "to_regex"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::bytes,
+            required: true,
+        }]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "regex",
+                source: "to_regex(s'^foobar$')",
+                result: Ok(r'^foobar$'),
+            }
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ToRegexFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ToRegexFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ToRegexFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        use Value::*;
+
+        let value = self.value.resolve(ctx)?.try_bytes()?;
+        let regex = Regex::new(value).map_err(|err| format!("could not create regex: {}",err))?;
+        Value::Regex(regex);
+
+        Ok(regex)
```

Comment:
```suggestion
        let value = self.value.resolve(ctx)?.try_bytes()?;
        let regex = regex::Regex::new(value).map_err(|err| format!("could not create regex: {}",err))?.map(Into:into);
```

This is a bit more concise. I realized I was unclear before too with my example. Having `Value::Regex(regex)` on its own line doesn't do anything. I had meant to assign that to a variable.